include/kernel.h

```diff
@@ -23,6 +23,11 @@
 extern "C" {
 #endif
 
+#ifdef CONFIG_BOOT_TIME_MEASUREMENT
```

Comment:
None of this is public API material, can we put this somewhere else other than kernel.h which is a public API header?

Reply:
I'm not entirely sure where to put it.  These variables are defined and manipulated from `kernel/`, and used in `tests/` (which probably should be `samples/`, how is this a test?), and are enabled by an option in `subsys/debug/`.  Whoever originally put this in seemed confused about its categorization, as well.

Previously these variables were simply redeclared in all these places, but I don't think going back to that is a good solution.

I'm open to suggestions.

Reply:
How about `kernel/include/kernel_internal.h` ?

Reply:
kernel_internal.h seems a good spot.

Reply:
I'm inclined to merge this as-is, I'm cleaning up how the arch interfaces are specified anyway and will take this under consideration.